<launch>
	<arg name="doDebug" default="false" />
	<arg if="$(arg doDebug)" name="launchPrefix" value="xterm -e gdb -args" />
	<arg unless="$(arg doDebug)" name="launchPrefix" value="" />

	<arg name="cmd_vel_topic_name" default="/cmd_vel" doc="cmd_vel topic name" />
	<arg name="cone_detector_topic_name" default="/cone_detector" doc="cone_detector topic name" />
	<arg name="image_topic_name" default="/usb_cam/image_raw" doc="topic name published by camera" />
	<arg name="odometry_topic_name" default="/odom" doc="odometry topic name" />
	<arg name="show_debug_windows" default="true" doc="true => show parameter window for cone detector" />
	
	<include file="$(find victoria_perception)/launch/cone_detector.launch">
		<arg name="image_transport" value="compressed" />
		<arg name="image_topic_name" value="$(arg image_topic_name)" />
		<arg name="show_debug_windows" value="$(arg show_debug_windows)" />
   	</include>

  <node pkg="victoria_navigation" name="cone_finder_strategy_node" type="cone_finder_strategy_node" respawn="false" output="screen" launch-prefix="$(arg launchPrefix)" >
  	<param name="cmd_vel_topic_name" value="$(arg cmd_vel_topic_name)" />
  	<param name="cone_detector_topic_name" value="$(arg cone_detector_topic_name)" />
  	<param name="odometry_topic_name" value="$(arg odometry_topic_name)" />
  </node>

</launch>
